Dockerfile

```diff
@@ -20,14 +20,18 @@ RUN git clone https://github.com/commonsense/conceptnet-puppet; \
 #   default docker configuration; it has no effect for building and running
 #   the conceptnet server
 RUN cd conceptnet-puppet; \
-	sh puppet-apply.sh; \
-	service postgresql stop
+        sh puppet-apply.sh
 
 COPY conceptnet.sh .
 
 RUN apt-get update && apt-get install -y postgresql-server-dev-10
 RUN cd /home/conceptnet/conceptnet5/web && /home/conceptnet/env/bin/pip3 install -e .
 
+USER postgres
+RUN wget https://tz-services-1.snet.sh/conceptnet-dumps/conceptnet5-pg-dump-20190502 -O /tmp/conceptnet5-pg-dump-20190502 && service postgresql start && pg_restore -h /var/run/postgresql/ -j20 --clean -U postgres --dbname=conceptnet5 /tmp/conceptnet5-pg-dump-20190502 || true && service postgresql stop && rm /tmp/conceptnet5-pg-dump-20190502
+
```

Comment:
The restore step should be a separate step. If one wants to rebuild the database or restore it, it should be their choice.

Reply:
One way to do it would be to have two dockerfiles(there might be other ways) but I opted to use this route as that is the general use case. I might be making mistake on that regard,

For instance; the steps to build the database is outlined by `bash conceptnet.sh build` in a private repo which i found to be intimately tied to that repo. 

So would you suggest the two dockerfile approaches for this repo with the intention to have clearly demarcated use cases?

Reply:
This will  break the use case were in a develper doesn't have the required resources for a restore or build. 

How about moving the logic of the above command to a new `restore-conceptnet-db` command in manage.py. This maintains the present workflow for developers, as well as enable deployment.

Reply:
Well; if a person doesn't have the resources to build conceptnet or restore, why build it? Or why have the docker container on their machine? 

Building conceptnet or restoring it requires about 80GB and signficant RAM, but if a person doesn't have those resources it would be better to skip it all together. Why keep it? We currently have a server that provides that functionality in our server. 

There is one case where this matters; if we have a test-conceptnet database that is a much much smaller set of conceptnet db. But i don't see a use-case for that. 

Is there any particular reason for doing this outside this repo? 

My aim here is: 
* Remove conceptnet docker build related commands from our scripts and use the server deployed version to interact with conceptnet. 
* For use cases where a person wants to have to build the conceptnet, for instance when they want to upgrade the conceptnet version defined in https://github.com/singnet/conceptnet5 and https://github.com/singnet/conceptnet-puppet, let them update the scripts there and update the image or *when they want to deploy* in a aws instance, use the aforementioned docker container with two Dockerfile(one to restore and another to **build**) to get the image which has the functionality we want.

Reply:
@tesYolan  I think we need both options: "build" and "restore". Our profiling results pointed to performance issues in CN queries (well, not exactly performance issues but the queries are taking more time than we'd like) so we'll probably need to prune the knowledge base we use to feed it.

We may want to select the topics by hand according to the use cases we want to implement and my first impression is that doing this in the text files is simpler than in the DB tables. We may even want to build different versions of the knowledge bases to implement different use cases. So we may need the flexibility to build from scratch.

Another point in favor of keeping the "build" option is because it would make easier to keep our CN up to date.

Reply:
@tesYolan I didnt' know you were intending to remove the `va_conceptnet_server` DockerComponent from manage.py. If that the case, then it doesn't matter for the use case I mentioned above. Please make a pr for that too, before merging this.

Reply:
@andre-senna I understand that requirement; how to implement it then? Currently the manage.py is the one responsible to building it. W/c is, IMHO, quite empty and doesn't really address all of the requirements you mentioned. It just builds the image with `bash conceptnet.sh build`. 

One thing that I can propose is have different dumps with those curated concepnets.
If there is a specific version with pruned KB, then we could make it available along side the dump that Amen has provided to make it faster for others. 

Then have a one docker file that builds it by pulling it from some server using environmental variables set name for the address. 

That way, we just have the responsibility of restore as that is what we mostly want. 

And please note; this version pegs the conceptnet5 and puppet installation scripts to a specific version cloned on singnet repo and modified.(Cause of the various annoy issues that i have discussed elsewhere). IF we want to update it, we should update singnet/conceptnet5 and singnet/conceptnet-puppet and test them using cli tools before we are integrating it. 

It might also make sense to pay for AMI on aws for conceptnet if our infra is also there, it might better for speed and avoid such installation hassels.

Reply:
If you think we can build the curated DBs easily without using manage.py then I'm OK with the change you suggest. My objection was just because I thought it would be easier to build those curated DBs using manage.py.

Reply:
At some point we will need to properly profile the CN server to figure out where the bottleneck is but I don't think we need to do it right now.